Expensify/App
# Place the new marker when a chat becomes visible

@tgolen, @marcaaron, since we were all working on this. 
  
### Details
I decided to put the code on `onVisibilityChange` in `ReportActionsView` because that way we keep all the "similar" logic in the same file and I think it is really easy to understand and follow what's going on.
BUT, it could totally go [here](https://github.com/Expensify/Expensify.cash/blob/main/src/libs/actions/Report.js#L519) in `updateReportWithNewAction` too and it works just as well and it is also a logical place for it to be so... 🤷. If anyone has a strong preference, I'm fine with either.

### Fixed Issues

https://github.com/Expensify/Expensify/issues/163167

### Tests/ QA Steps
In web:
- Open a chat that does NOT have the New marker in place and move to a different tab.
- Receive two messages in the chat (wait for the notifications)
- Go back to the `.cash` tab. The new marker should be over the oldest of the new messages.
- Open a chat that DOES have the new marker (or place it marking a comment as unread) and move to a different tab.
- Receive a message in the chat (wait for the notification)
- Go back to the `.cash` tab. The New marker should be in the same place it was before.

In web and all the other platforms:
- Open a chat that does NOT have the New marker in place and minimize the window/app.
- Receive two messages in the chat (wait for the notifications)
- Go back to the `.cash` tab. The new marker should be over the oldest of the new messages.
- Open a chat that DOES have the new marker (or place it marking a comment as unread) and minimize the window/app.
- Receive a message in the chat (wait for the notification)
- Go back to the `.cash` tab. The New marker should be in the same place it was before.

### Tested On


### Screenshots
Taking screenshots of the marker won't mean much.
